packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -619,15 +619,12 @@ function createArrayNodeProxy(
 			const field = getSequenceField(proxy);
 			const maybeIndex = asIndex(key, field.length);
 			if (maybeIndex !== undefined) {
-				const val = field.boxedAt(maybeIndex);
+				const val = field.at(maybeIndex);
 				// To satisfy 'deepEquals' level scrutiny, the property descriptor for indexed properties must
 				// be a simple value property (as opposed to using getter) and declared writable/enumerable/configurable.
 				return {
-					// TODO: Ideally, we would return leaves without first boxing them.  However, this is not
```

Comment:
since field nodes haven't done implicit unboxing for a while, this TODO didn't apply, and I found it when searching for references to field nodes.

Removing this TODO and thus optimizing this code was the simplest way to deal with this out of date comment: access to monomorphic leaf arrays is likely faster and uses less memory now.